tally/include/tally/counter.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#pragma once
+
+#include <cstdint>
+
+namespace tally {
+
+class Counter {
+ public:
+  virtual ~Counter() = default;
+
+  // Increments the counter by 1.
+  virtual void Inc() = 0;
+
+  // Increments the counter by `value`.
+  virtual void Inc(int64_t value) = 0;
```

Comment:
method overloading, how i've missed thee 👍

Reply:
We actually did this work already in https://github.com/jaegertracing/jaeger-client-cpp/tree/master/src/jaegertracing/metrics. Reuse what you like.

Reply:
Nice! I wish I'd known about this before lol.